backend/src/apiserver/storage/pipeline_store_kubernetes.go

```diff
@@ -0,0 +1,508 @@
+	if err != nil {
+		return err
+	}
+
+	conditionSet := false
+
+	for i := range k8sPipelineVersion.Status.Conditions {
+		condition := &k8sPipelineVersion.Status.Conditions[i]
+
+		if condition.Type == "PipelineVersionStatus" {
+			if condition.Reason == string(status) && condition.Message == condition.Reason {
+				return nil
+			}
+
+			condition.Reason = string(status)
+			condition.Message = string(status)
+			condition.Status = metav1.ConditionTrue
+
+			conditionSet = true
+
+			break
+		}
+	}
+
+	if !conditionSet {
+		k8sPipelineVersion.Status.Conditions = append(k8sPipelineVersion.Status.Conditions, v2beta1.SimplifiedCondition{
+			Type:    "PipelineVersionStatus",
+			Reason:  string(status),
+			Message: string(status),
+			Status:  metav1.ConditionTrue,
+		})
+	}
+
+	err = k.client.Status().Update(context.TODO(), k8sPipelineVersion)
+	if err != nil {
+		return util.NewInternalServerError(err, "Failed to update the pipeline version status")
+	}
+
+	return nil
+}
+
+func (k *PipelineStoreKubernetes) CreatePipelineVersion(pipelineVersion *model.PipelineVersion) (*model.PipelineVersion, error) {
+	pipeline, err := k.GetPipeline(pipelineVersion.PipelineId)
+	if err != nil {
+		return nil, err
+	}
+
+	return k.createPipelineVersionWithPipeline(pipeline, pipelineVersion)
+}
+
+func (k *PipelineStoreKubernetes) UpdatePipelineDefaultVersion(pipelineId string, versionId string) error {
+	// Default version is deprecated and was a KFPv1 thing. In KFPv2, we should not support. Reference:
+	// https://github.com/kubeflow/pipelines/blob/d1b15ef4da33cbeafa491564318c7e2a68dc431f/backend/src/apiserver/resource/resource_manager.go#L329-L334
+	return util.NewBadRequestError(errors.New("pipeline default version is unsupported"),
+		"pipeline default version is unsupported when storing in Kubernetes")
+}
+
+func (k *PipelineStoreKubernetes) GetLatestPipelineVersion(pipelineId string) (*model.PipelineVersion, error) {
+	k8sPipelineVersions, err := k.getK8sPipelineVersions()
+	if err != nil {
+		return nil, err
+	}
+
+	var latestK8sPipelineVersion *v2beta1.PipelineVersion
+
+	sort.SliceStable(k8sPipelineVersions.Items, func(i, j int) bool {
+		return k8sPipelineVersions.Items[i].CreationTimestamp.After(k8sPipelineVersions.Items[j].CreationTimestamp.Time)
+	})
+	latestK8sPipelineVersion = &k8sPipelineVersions.Items[0]
+
+	return latestK8sPipelineVersion.ToModel()
+}
+
+func (k *PipelineStoreKubernetes) GetPipelineVersion(pipelineVersionId string) (*model.PipelineVersion, error) {
+	pipelineVersion, err := k.getK8sPipelineVersion(pipelineVersionId)
+	if err != nil {
+		return nil, err
+	}
+
+	return pipelineVersion.ToModel()
+}
+
+func (k *PipelineStoreKubernetes) GetPipelineVersionByName(name string) (*model.PipelineVersion, error) {
+	k8sPipelineVersions, err := k.getK8sPipelineVersions()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, k8sPipelineVersion := range k8sPipelineVersions.Items {
+		if k8sPipelineVersion.Name == name {
```

Comment:
I don't think this is safe because you are going across namespaces when in multiuser mode so many of these could exist. @HumairAK , should this default to the pod/kubeflow namespace when in multiuser mode? The only place I saw this used was in loading the sample pipelines.